packages/frontend-2/components/viewer/models/Card.vue

```diff
@@ -410,4 +414,69 @@ const onActionChosen = (params: { item: LayoutMenuItem }) => {
       break
   }
 }
+
+const expandToShowSelectedObjects = (selectedIds: string[]) => {
+  // Find paths to all selected objects in this model's tree
+  const pathsToExpand = findPathsToSelectedObjects(selectedIds)
+
+  if (pathsToExpand.length > 0) {
+    // Expand the model itself
+    if (!isExpanded.value) {
+      isExpanded.value = true
+    }
+
+    // Clear previous force-expanded nodes and set new ones
+    forceExpandedNodeIds.value.clear()
+    const nodeIdsToExpand = pathsToExpand.flat()
+    nodeIdsToExpand.forEach((id) => forceExpandedNodeIds.value.add(id))
+  }
+}
+
+const findPathsToSelectedObjects = (selectedIds: string[]): string[][] => {
+  const paths: string[][] = []
+
+  for (const rootNode of props.rootNodes) {
+    const nodePaths = findPathsInNode(rootNode, selectedIds, [])
+    paths.push(...nodePaths)
+  }
+
+  return paths
+}
+
+const findPathsInNode = (
+  node: ExplorerNode,
+  selectedIds: string[],
+  currentPath: string[]
+): string[][] => {
+  const paths: string[][] = []
+  const nodeId = node.raw?.id || node.guid || ''
+  const newPath = nodeId ? [...currentPath, nodeId] : currentPath
+
+  if (nodeId && selectedIds.includes(nodeId)) {
+    paths.push(newPath)
+  }
+  if (node.children) {
+    for (const child of node.children) {
+      const childPaths = findPathsInNode(child, selectedIds, newPath)
+      paths.push(...childPaths)
+    }
+  }
+
+  return paths
+}
+
+// Auto-expand tree to show selected objects
+watch(
+  () => objects.value,
+  (selectedObjects) => {
+    const selectedIds = selectedObjects.map((obj: { id: string }) => obj.id)
+
+    if (selectedObjects.length > 0) {
+      expandToShowSelectedObjects(selectedIds)
+    } else {
+      forceExpandedNodeIds.value.clear()
+    }
+  },
+  { deep: true }
```

Comment:
this is a bit concerning, i assume that the objects can be pretty big/deep at times. is a shallow watcher not enough here? selectedObjects should update every time something gets added or removed anyway